src/util/symbol_table_base.h

```diff
@@ -117,6 +117,78 @@ class symbol_table_baset
   virtual void clear() = 0;
 
   void show(std::ostream &out) const;
+
+  class iteratort
```

Comment:
Shouldn't iterators inherit some tag class to indicate they're a forward, bidir, random-access iterator etc? Or does inheriting category below achieve this?

Reply:
Correct, that's what the `iterator_category` is.